libyul/optimiser/OptimizerUtilities.cpp

```diff
@@ -36,6 +36,16 @@ using namespace solidity::langutil;
 using namespace solidity::util;
 using namespace solidity::yul;
 
+namespace
+{
+
+bool hasLeadingOrTrailingDot(std::string_view const _s)
+{
+	return !_s.empty() && (_s.at(0) == '.' || _s.at(_s.size()-1) == '.');
```

Comment:
```suggestion
	return !_s.empty() && (_s.front() == '.' || _s.back() == '.');
```

Also, *technically*, `isRestrictedIdentifier` already checks for said empty identifier and short circuits in such a case, so I would go even further to reduce this and omit the empty check completely, and rather replace it with an assert.

```suggestion
	solAssert(!_s.empty());
	return _s.front() == '.' || _s.back() == '.';
```